src/include/frag.h

```diff
@@ -579,7 +579,6 @@ class fragset_t {
   }
   void insert(frag_t f) {
     _set.insert(f);
-    simplify();
```

Comment:
Could you explain why happens in simplify() that causes the frags to be missing on the replicas.

Reply:
`simplify()` computes the bit representation of the least common ancestor of the frags.
When this representation is forwarded to the replicas, the `ScrubStack::handle_scrub()` method in the `OP_QUEUEDIR` case simply searches this simplified representation in the list of (unsimplified) frags ... which it fails to find.
Hence we get to see the `no frag 10*` type of logs.

Reply:
I can understand removing `simplify()` in the `ScrubStack` code but is this point also necessary?

Reply:
The fragset is simplified at every insert. Haven't tested this specific change. I presumed that since the fragset is being constructed the scrub path, any simplification would hit the same case.

I'll specifically test without this change and get back.